src/e2e/java/uk/gov/hmcts/reform/sscs/functional/sya/notifications/DocmosisWithGovNotifyLetterFunctionalTest.java

```diff
@@ -1,40 +1,91 @@
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 import junitparams.JUnitParamsRunner;
 import junitparams.Parameters;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import uk.gov.hmcts.reform.sscs.domain.notify.NotificationEventType;
 import uk.gov.hmcts.reform.sscs.functional.AbstractFunctionalTest;
 import uk.gov.service.notify.Notification;
-import uk.gov.service.notify.NotificationClientException;
 
 @RunWith(JUnitParamsRunner.class)
 public class DocmosisWithGovNotifyLetterFunctionalTest extends AbstractFunctionalTest {
 
+    public static final String EXPECTED_LETTER_SUBJECT = "Pre-compiled PDF";
+    //TODO: Add callback jsons for these letters to test them functionally
+    private static final Set<NotificationEventType> DOCMOSIS_LETTERS_WITH_NO_TEST_CALLBACK = EnumSet.of(
+        ACTION_HEARING_RECORDING_REQUEST,
+        ACTION_POSTPONEMENT_REQUEST_WELSH,
+        ADMIN_APPEAL_WITHDRAWN,
+        APPEAL_LAPSED_NOTIFICATION,
+        APPEAL_WITHDRAWN_NOTIFICATION,
+        DECISION_ISSUED_WELSH,
+        DIRECTION_ISSUED_WELSH,
+        DWP_APPEAL_LAPSED_NOTIFICATION,
+        DWP_UPLOAD_RESPONSE_NOTIFICATION,
+        HMCTS_APPEAL_LAPSED_NOTIFICATION,
+        JOINT_PARTY_ADDED,
+        POSTPONEMENT_NOTIFICATION,
+        RESEND_APPEAL_CREATED_NOTIFICATION,
+        VALID_APPEAL_CREATED
+    );
+
     public DocmosisWithGovNotifyLetterFunctionalTest() {
         super(30);
     }
 
     @Test
     @Parameters(method = "eventTypes")
-    public void shouldSendDocmosisLetters(NotificationEventType notificationEventType, int expectedNumberOfLetters) throws IOException, NotificationClientException {
+    public void shouldSendDocmosisLetters(NotificationEventType notificationEventType) {
+        simulateCcdCallback(notificationEventType);
+
+        assertThatNoException().isThrownBy(() -> {
+            List<Notification> notifications = fetchLetters();
+            assertThat(notifications)
+                .extracting(Notification::getSubject)
+                .allSatisfy(subject ->
+                    assertThat(subject)
+                        .isPresent()
+                        .hasValue(EXPECTED_LETTER_SUBJECT)
+                );
+        });
+    }
 
-        simulateCcdCallback(notificationEventType,
-                notificationEventType.getId() + "Callback.json");
+    @Test
+    @Parameters(method = "expectedNumberOfLetters")
+    public void shouldSendCorrectNumberOfDocmosisLetters(NotificationEventType notificationEventType,
+                                                         int expectedNumberOfLetters) {
+        simulateCcdCallback(notificationEventType);
 
-        List<Notification> notifications = fetchLetters();
-        assertEquals(expectedNumberOfLetters, notifications.size());
-        notifications.forEach(n -> assertEquals("Pre-compiled PDF", n.getSubject().orElse("Unknown Subject")));
+        assertThatNoException().isThrownBy(() -> {
```

Comment:
Same as above on personal taste on the use of `assertThatNoException`.

Reply:
Removed